src/backend/distributed/executor/local_executor.c

```diff
@@ -506,3 +505,72 @@ DisableLocalExecution(void)
 					  (superuser() ? PGC_SUSET : PGC_USERSET), PGC_S_SESSION,
 					  GUC_ACTION_LOCAL, true, 0, false);
 }
+
+
+void
+SetTaskQueryAndPlacementList(Task *task, Query *query, List *placementList)
+{
+	task->taskPlacementList = placementList;
+
+	if (TaskAccessesLocalNode(task))
+	{
+		task->query = query;
+		task->queryStringLazy = NULL;
+		return;
+	}
+
+	task->query = NULL;
+	MemoryContext previousContext = MemoryContextSwitchTo(GetMemoryChunkContext(task));
+	StringInfo queryString = makeStringInfo();
+
+	pg_get_query_def(query, queryString);
+
+	task->queryStringLazy = queryString->data;
+	MemoryContextSwitchTo(previousContext);
+}
+
+
+/*
+ * TaskQueryString generates task->queryStringLazy if missing.
+ *
+ * For performance reasons, the queryString is generated lazily. For example
+ * for local queries it is usually not needed to generate it, so this way we
+ * can skip the expensive deparsing+parsing.
+ */
+char *
+TaskQueryString(Task *task)
```

Comment:
It seems more appropriate to have this function (and maybe SetTaskQueryAndPlacementList) in deparse_shard_query.c, since it's more of general deparsing function.

Reply:
Makes sense, moved it there